dial.go

```diff
@@ -0,0 +1,143 @@
+package tlsdialer
+
+import (
+	"bufio"
+	"context"
+	"errors"
+	"net"
+	"os"
+	"strings"
+
+	"github.com/tarantool/go-openssl"
+)
+
+func sslDialContext(ctx context.Context, network, address string,
+	sslOpts opts) (connection net.Conn, err error) {
+	var sslCtx interface{}
+	if sslCtx, err = sslCreateContext(sslOpts); err != nil {
+		return
+	}
+
+	return openssl.DialContext(ctx, network, address, sslCtx.(*openssl.Ctx), 0)
```

Comment:
After the previous fix:
```suggestion
	return openssl.DialContext(ctx, network, address, sslCtx, 0)
```